mne/io/bti/bti.py

```diff
@@ -1063,7 +1063,13 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
                 block.shape = shape
                 data_view = data[:, sample_start:sample_stop]
                 one = np.empty(block.shape[::-1])
-                for ii, b_i_o in enumerate(bti_info['order']):
+
+                order = bti_info['order']
+                if order is None:
+                    order_ = np.arange(block.shape[1])
+                else:
+                    order_ = bti_info['order']
+                for ii, b_i_o in enumerate(order_):
```

Comment:
I was just pointing that what you suggest is a bit different in behavior :)

but whatever @dengemann agrees on is fine with me.